src/share/poudriere/jail.sh

```diff
@@ -273,6 +277,11 @@ update_jail() {
 		err 1 "Unsupported method"
 		;;
 	esac
+
+	if [ ${XDEV} -eq 1 ]; then
```

Comment:
This comes too late, note line 261 where the jail is marked clean.  It's the `@clean` zfs snapshot which is rolled back to at startup.  You'll need to hook `install_xdev_standalone` from where it used to be called.